cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/LogHandlerTestRun.java

```diff
@@ -474,71 +384,70 @@ private void testLogs(String appId, String entityType, String entityId, String n
-    Assert.assertEquals(15, logLines.size());
-    int expected = 20;
-    for (String log : logLines) {
-      String expectedStr = entityId + "&lt;img&gt;-" + expected;
-      Assert.assertEquals(expectedStr, log.substring(log.length() - expectedStr.length()));
-      expected++;
-    }
+    verifyLogs(response, entityId, "text", false, true, true, 15, 20);
 
     // Try with invalid time range -> start > stop
     logsUrl = String.format("apps/%s/%s/%s/logs?start=%s&stop=%s",
-      appId, entityType, entityId, 350, 300);
+                            appId, entityType, entityId, 350, 300);
     response = doGet(getVersionedAPIPath(logsUrl, namespace));
     Assert.assertEquals(HttpResponseStatus.BAD_REQUEST.getCode(), response.getStatusLine().getStatusCode());
   }
 
-  private void testLogsRunId(String appId, String entityType, String entityId, String namespace) throws Exception {
-    Id.Program id = Id.Program.from(namespace, appId, ProgramType.valueOfCategoryName(entityType), entityId);
-    RunRecord runRecord = mockLogReader.getRunRecord(id);
-    int expectedEvents = 20;
-    if (runRecord.getStatus() == ProgramRunStatus.RUNNING || runRecord.getStatus() == ProgramRunStatus.SUSPENDED) {
-      expectedEvents = 30;
-    }
-    long startTime = MockLogReader.getMockTimeSecs(0);
-    long stopTime = MockLogReader.getMockTimeSecs(100);
-    String nextNoFromUrl = String.format("apps/%s/%s/%s/runs/%s/logs?start=%s&stop=%s",
-                                         appId, entityType, entityId, runRecord.getPid(), startTime, stopTime);
-    HttpResponse response = doGet(getVersionedAPIPath(nextNoFromUrl, namespace));
-    Assert.assertEquals(HttpResponseStatus.OK.getCode(), response.getStatusLine().getStatusCode());
-    String out = EntityUtils.toString(response.getEntity());
-    List<String> logLines = Lists.newArrayList(Splitter.on("\n").omitEmptyStrings().split(out));
-    Assert.assertEquals(expectedEvents, logLines.size());
-    int expected = 20;
-    for (String log : logLines) {
-      String expectedStr = entityId + "&lt;img&gt;-" + expected;
-      Assert.assertEquals(expectedStr, log.substring(log.length() - expectedStr.length()));
-      expected += 2;
-    }
-  }
-
   private void testLogsFilter(String appId, String entityType, String entityId, String namespace) throws Exception {
     long startTime = MockLogReader.getMockTimeSecs(20);
     long stopTime = MockLogReader.getMockTimeSecs(35);
     String logsFilterUrl = String.format("apps/%s/%s/%s/logs?start=%s&stop=%s&filter=loglevel=ERROR", appId,
                                          entityType, entityId, startTime, stopTime);
     HttpResponse response = doGet(getVersionedAPIPath(logsFilterUrl, namespace));
+    verifyLogs(response, entityId, "text", true, true, true, 8, 20);
+  }
+
+  private void verifyLogs(HttpResponse response, String entityId, String format, boolean runIdOrFilter,
```

Comment:
+1 for the refactor. 

It would be good to have javadoc for this method, and some comments in the method itself.